here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/AbstractTask.java

```diff
@@ -54,16 +51,23 @@ public abstract class AbstractTask<RESULT, SELF extends AbstractTask<RESULT, SEL
     implements INakshaBound, UncaughtExceptionHandler {
 
   private static final Logger log = LoggerFactory.getLogger(AbstractTask.class);
-  static int thresholdPerProcessor;
-  static int percentageThresholdPerPrinciple;
+  private static int maxParallelRequestsPerCPU = 50;
+  private static int maxPctParallelRequestsPerPrincipal = 100;
 
   public static void initConcurrencyLimits(int cpuThreshHold, int perPrincipalThreshold) {
```

Comment:
This function is only setting the threshold values, but not calculating the target limits (`limit` and `principalLimit`). How is this scenario working or tested?

Reply:
The function is static and not thread safe (nor are the variables being written to). I'am generally fine with this, but please document it, and add a big fat warning, that the methods MUST NOT be invoked concurrently, other solution is to make the init method thread safe (question is, if it really needs to be, because it will slow down reading the variables and I see no benefit)

Reply:
Fixed.